master/custom_steps.py

```diff
@@ -130,6 +131,26 @@ def run(self):
         return status
 
 
+# Like FileUpload, but if the dest file already exists,
+# just log that to stdio and do nothing. Useful when the
+# filename contains (eg) a git commit or SHA that uniquely
+# identifies the file version.
+class FileUploadIfNotExist(FileUpload):
+    name = 'file-upload-if-not-exist'
+
+    @defer.inlineCallbacks
+    def run(self):
+        masterdest = os.path.expanduser(self.masterdest)
+        if os.path.isfile(masterdest) and os.path.getsize(masterdest) > 0:
+            stdio = yield self.addLog('stdio')
+            stdio.addStdout(f"File {repr(masterdest)} already exists on dest, skipping upload!")
+            yield stdio.finish()
+            return SUCCESS
+
+        yield from super().run()
+        return SUCCESS
```

Comment:
```suggestion
        res = yield super().run()
        return res
```
Can you humor me and try this variant? I'm just worried about suppressing failures from the super-run step.